src/hexer/destroyer.nim

```diff
@@ -204,10 +204,10 @@ proc trProcDecl(c: var Context; n: var Cursor) =
   c.dest.addParRi()
 
 proc trNestedScope(c: var Context; body: var Cursor; kind = Other) =
-  var bodyScope = createNestedScope(kind, c.currentScope, body.info)
-  swap c.currentScope, bodyScope
```

Comment:
Because `Scope` is a value, it formed a self-cycle because `c.parent` always points to `c.currentScope`